borsh-derive-internal/src/enum_de.rs

```diff
@@ -77,7 +77,7 @@ pub fn enum_de(input: &ItemEnum, cratename: Ident) -> syn::Result<TokenStream2>
     if let Some(method_ident) = init_method {
         Ok(quote! {
             impl #impl_generics #cratename::de::BorshDeserialize for #name #ty_generics #where_clause {
-                fn deserialize(buf: &mut &[u8]) -> core::result::Result<Self, #cratename::maybestd::io::Error> {
+                fn deserialize(buf: &mut &[u8]) -> ::core::result::Result<Self, #cratename::maybestd::io::Error> {
```

Comment:
Ideally, we'd use a name from `borsh` crate here, like `serde` does, but `::core` is already an improvement: 

https://github.com/serde-rs/serde/blob/ebd06eebdbbacf0c2d5beeb6ebddc77bb2e37bd4/serde_derive/src/ser.rs#L36

Reply:
And that would keep sanity for if you imported a crate named `core`? That one feels more opinionated since it increases the exposed `__private` API so maybe best to do it in a separate PR?

Reply:
That's definitely should be done in a separate PR, if at all. 

Practically, I wouldn't expect this to make any difference, but it's just good hygiene (pun intended) to only rely on names provided by the runtime-support crate and not rely on any properties of the call-site environment.